addons/hr/models/mail_channel.py

```diff
@@ -11,6 +12,14 @@ class Channel(models.Model):
         'hr.department', string='HR Departments',
         help='Automatically subscribe members of those departments to the channel.')
 
+    # CONSTRAINTS
+
+    @api.constrains('subscription_department_ids')
+    def _constraint_subscription_department_ids_channel(self):
+        failing_channel = self.sudo().filtered(lambda channel: channel.channel_type!='channel' and channel.subscription_department_ids)
+        if failing_channel:
+            raise ValidationError(_("The following channels have department auto-subscription but are not of right type 'channel': %(channel_names)s"), channel_names = ', '.join(failing_channel.mapped('name')))
```

Comment:
`but are not of right type 'channel'` reads a bit strange. Either we should say they are not the right type in generic terms, or we should say what is the right type, but mixing both on the same expression looks incorrect. Maybe just using parenthesis instead of quote around channel would make it better.

Also note that channel name can actually be empty (in case of group with auto-name based on members), so this error might make no sense with just empty separated by `,`.